subdomains/blog/_posts/2020-07-21-declarative-tools.md

```diff
@@ -0,0 +1,29 @@
+---
+layout: post
+title:  "Bringing the Declarative Experience to Your Tools"
+date:   2020-07-17 08:00:00 -0700
+categories: [ux, delight]
+author: Chuck Pierce
+excerpt_separator: <!--more-->
+---
+
+Declarative UI—powered by packages like React, Vue, or Angular—has completely taken over the world. These techniques let you focus on describing how an app should look and feel, without having to micromanage making each change happen imperatively. Despite that, managing the tools you need for development is often stuck in an imperative world. In the same way that modern frameworks made it easier to build declarative UIs, **Volta brings the power of the declarative paradigm to your tools!**
```

Comment:
"Volta brings the power of the declarative paradigm to your tools" is ambiguous. When I first read it, I thought you were saying "Volta is a declarative tool" which doesn't meaningfully differentiate it from, say, React, Vue, and Angular in your earlier examples. I think what you're saying is it brings it to "the experience of managing your tools."

I think the problem is people don't really have an innate intuition for "toolchain" or "managing tools." They don't really know what that is. I think a more intuitive concept is **installing tools**. Everyone installs stuff, and nobody particularly likes doing it. Worse, it's always a little anxiety-inducing because you never know if it's going to change the state of your system in some irreversible way.

I think a more punchy and visceral way to say this would be something like "declare your tools, don't install them." Or "stop installing your tools and start declaring them." And I wonder if the more visceral value proposition is, not "micromanage making each change happen imperatively" but rather "completely liberate you from worrying about the state of your system."

Reply:
Reading on, the rest of the details later in the post are good benefits as well. I think all of them have to do with keeping track of the state of your system in your head.

- Don't have to remember if you've installed the right version of tools
- Don't have to worry about them making some change to your system that'll be hard to undo
- Don't have to remember to switch between versions

All of this is about liberating the user from the cognitive burden of remembering the state.